sonar-db/src/main/java/org/sonar/db/version/v564/CleanUsurperRootComponents.java

```diff
@@ -0,0 +1,156 @@
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select" +
+      " sn.id,p.scope,p.qualifier" +
+      " from" +
+      " snapshots sn, projects p" +
+      " where" +
+      " p.uuid = sn.component_uuid" +
+      " and (p.qualifier<>sn.qualifier or p.scope<>sn.scope)");
+    massUpdate.update("update snapshots set scope=?,qualifier=? where id=?");
+    massUpdate.rowPluralName("snapshots with inconsistent scope or qualifier");
+    massUpdate.execute((row, update) -> {
+      long snapshotId = row.getLong(1);
+      String scope = row.getString(2);
+      String qualifier = row.getString(3);
+
+      update.setString(1, scope);
+      update.setString(2, qualifier);
+      update.setLong(3, snapshotId);
+
+      return true;
+    });
+  }
+
+  private static void cleanUsurperRootComponents(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select p.id,p.uuid from projects p " +
+      " where" +
+      " p.project_uuid = p.uuid" +
+      " and not (" +
+      " p.scope = 'PRJ'" +
+      " and p.qualifier in ('TRK', 'VW', 'DEV')" +
+      " )");
+    massUpdate.update("delete from duplications_index where snapshot_id in (select id from snapshots where component_uuid=?)");
+    massUpdate.update("delete from project_measures where component_uuid=?");
+    massUpdate.update("delete from ce_activity where component_uuid=?");
+    massUpdate.update("delete from events where component_uuid=?");
+    massUpdate.update("delete from project_links where component_uuid=?");
+    massUpdate.update("delete from snapshots where component_uuid=? or root_component_uuid=?");
+    massUpdate.update("delete from issues where component_uuid=? or project_uuid=?");
+    massUpdate.update("delete from file_sources where file_uuid=? or project_uuid=?");
+    massUpdate.update("delete from group_roles where resource_id=?");
+    massUpdate.update("delete from user_roles where resource_id=?");
+    massUpdate.update("delete from properties where resource_id=?");
+    massUpdate.update("delete from widgets where resource_id=?");
+    massUpdate.update("delete from projects where uuid=?");
+    massUpdate.rowPluralName("usurper root components");
+    massUpdate.execute((row, update, updateIndex) -> {
+      long componentId = row.getLong(1);
+      String componentUuid = row.getString(2);
+      switch (updateIndex) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+          update.setString(1, componentUuid);
+          return true;
+        case 5:
+        case 6:
+        case 7:
+          update.setString(1, componentUuid);
+          update.setString(2, componentUuid);
+          return true;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+          update.setLong(1, componentId);
+          return true;
+        case 12:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This case's code block is the same as the block for the case on line 94. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1871)